### PR TITLE
Balloon block editor toolbar button position calculation fix.

### LINKED ISSUE #15045: Wrong position of Balloon Block Toolbar button.
## 📝 Provide detailed reproduction steps (if any)

1.  Open https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/balloon-block-editor.html
2.  Play with the demo

### ✔️ Expected result

_Toolbar button will be aligned with selected block content._

### ❓ Possible solution

_Wrongly calculation of button destination coordinates (ancestors with position: relative without scroll are summed - should not)_

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: …
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal (ui): Properly positioned toolbar button in Balloon block editor. Closes #15045.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-ui/tests/manual/blocktoolbar/blocktoolbar.html [hunk 1]
```diff
@@ -1,45 +1,75 @@
...
+						cultures online or from books, nothing comes close to experiencing cultural diversity in person.
+						You learn to appreciate each and every single one of the differences while you become more culturally fluid.
+					</p>
+
+					<h3>Improvisation</h3>
+					<p>
+						Life doesn't allow us to execute every single plan perfectly. This especially seems to be the case when
+						you travel. You plan it down to every minute with a big checklist. But when it comes to executing it,
+						something always comes up and you’re left with your improvising skills. You learn to adapt as you go.
+						Here’s how my travel checklist looks now:
```
Comment (arkflpc) [inline packages/ckeditor5-ui/tests/manual/blocktoolbar/blocktoolbar.html:32]: Why is it outside of any editor?

File: packages/ckeditor5-ui/tests/manual/blocktoolbar/blocktoolbar.html [hunk 2]
```diff
@@ -1,45 +1,75 @@
...
+					</p>
```
Comment (arkflpc) [inline packages/ckeditor5-ui/tests/manual/blocktoolbar/blocktoolbar.html:33]: Why is it outside any editor?

Comment (pszczesniak) [reply packages/ckeditor5-ui/tests/manual/blocktoolbar/blocktoolbar.html]: Explained here :point\_right: 35f3bb1#diff-35e1879959ec250ef809949866751a1df0684719286b0b91c9c10b131409eac7

File: packages/ckeditor5-utils/src/dom/rect.ts [hunk 3]
```diff
@@ -576,7 +576,7 @@ function getElementOverflow( element: HTMLElement | Range ): string {
  * (and scrollable) ancestors.
  */
 function shiftRectToCompensatePositionedAncestor( rect: Rect, positionedElementAncestor: HTMLElement ): void {
-	const ancestorPosition = new Rect( positionedElementAncestor );
```
Comment (niegowski) [inline packages/ckeditor5-utils/src/dom/rect.ts:579]: This change looks like it's not related and as we previously debugged it seems invalid.

File: packages/ckeditor5-utils/tests/manual/position/position.html [hunk 4]
```diff
@@ -2,13 +2,13 @@
 	<div class="test-box" style="height: 400px; width: 400px">
 		<div class="test-box" style="height: 800px; width: 800px">
 			<div class="target"></div>
-			<div class="source source-se"></div>
 		</div>
-		<div class="source source-sw"></div>
 	</div>
-	<div class="source source-ne"></div>
```
Comment (niegowski) [inline packages/ckeditor5-utils/tests/manual/position/position.html:9]: This seems not related and I'd not change this.

**Review by niegowski — CHANGES_REQUESTED** [review]
